packages/frontend/src/components/attachment/messageAttachment.tsx

```diff
@@ -235,37 +237,17 @@ export function DraftAttachment({
 }: {
   attachment: MessageTypeAttachmentSubset
 }) {
-  const [webxdcInfo, setWebxdcInfo] = useState<T.WebxdcMessageInfo | null>(null)
-  const [isLoadingWebxdcInfo, setIsLoadingWebxdcInfo] = useState(true)
-  const accountId = selectedAccountId()
-
-  const lastFileNameRef = useRef<string | null>(null)
-
-  useEffect(() => {
-    if (attachment.viewType === 'Webxdc') {
-      // Only load webxdc info if filename has changed
-      if (attachment.fileName !== lastFileNameRef.current) {
-        lastFileNameRef.current = attachment.fileName
-        setIsLoadingWebxdcInfo(true)
-        BackendRemote.rpc
-          .getWebxdcInfo(accountId, attachment.id)
-          .then((info: T.WebxdcMessageInfo) => {
-            setWebxdcInfo(info)
-          })
-          .catch((error: any) => {
-            console.error(
-              'Failed to load webxdc info for draft:',
-              attachment.id,
-              error
-            )
-            setWebxdcInfo(null)
-          })
-          .finally(() => {
-            setIsLoadingWebxdcInfo(false)
-          })
-      }
-    }
-  }, [accountId, attachment.id, attachment.viewType, attachment.fileName])
+  const isViewTypeWebxdc = attachment.viewType === 'Webxdc'
+  const [attachmentIdToLoad, setAttachmentIdToLoad] = useState(attachment.id)
+  if (useHasChanged2(attachment.fileName)) {
```

Comment:
```suggestion
  const hasFileNameChanged = useHasChanged2(attachment.fileName)
  if (hasFileNameChanged) {
```

to make it extra clear that rules of hooks are not violated (no conditional hooks).

Reply:
> to make it extra clear that rules of hooks are not violated (no conditional hooks).

IDK, those are caught at compile time. The fact that build succeeds means that there is no such error, so there is no need for this I'd say.

Reply:
Alright, if just for the sake of clarity then let's apply.